### PR TITLE
k8s libs: upgrade to v0.20.5

### LINKED ISSUE #10682: k8s libs: version upgrade
currently, we use v0.17.3 of k8s libs and we should upgrade to a newer version - try with the current "stable" v0.20.4, or, if that doesn't work - try with v0.19.8, etc.

### PR DESCRIPTION
**Pull request by prezha**
fixes #10682 

update: v0.20.5 was released 5 days ago, so the libraries are updated accordingly

try to upgrade k8s libs to the current latest stable v0.20.~4~5

locally-run version upgrade tests passed:

```
❯ env TEST_ARGS="-minikube-start-args=--driver=docker -test.run TestRunningBinaryUpgrade" make integration 
go test -ldflags="-X k8s.io/minikube/pkg/version.version=v1.18.0-beta.0 -X k8s.io/minikube/pkg/version.isoVersion=v1.18.0-beta.0 -X k8s.io/minikube/pkg/version.gitCommitID="4903ede887ea788fabe6eb3e9376bb6b5c34f5ba-dirty" -X k8s.io/minikube/pkg/version.storageProvisionerVersion=v4" -v -test.timeout=90m ./test/integration --tags="integration " -minikube-start-args=--driver=docker -test.run TestRunningBinaryUpgrade 2>&1 | tee "./out/testout_4903ede88.txt"
Found 16 cores, limiting parallelism with --test.parallel=9
=== RUN   TestRunningBinaryUpgrade
=== PAUSE TestRunningBinaryUpgrade
=== CONT  TestRunningBinaryUpgrade
    version_upgrade_test.go:114: (dbg) Run:  /tmp/minikube-v1.9.0.835108460.exe start -p running-upgrade-20210302000143-3349650 --memory=2200 --vm-driver=docker
    version_upgrade_test.go:114: (dbg) Done: /tmp/minikube-v1.9.0.835108460.exe start -p running-upgrade-20210302000143-3349650 --memory=2200 --vm-driver=docker: (48.502517798s)
    version_upgrade_test.go:124: (dbg) Run:  out/minikube start -p running-upgrade-20210302000143-3349650 --memory=2200 --alsologtostderr -v=1 --driver=docker
    version_upgrade_test.go:124: (dbg) Done: out/minikube start -p running-upgrade-20210302000143-3349650 --memory=2200 --alsologtostderr -v=1 --driver=docker: (41.764365879s)
    helpers_test.go:171: Cleaning up "running-upgrade-20210302000143-3349650" profile ...
    helpers_test.go:174: (dbg) Run:  out/minikube delete -p running-upgrade-20210302000143-3349650
    helpers_test.go:174: (dbg) Done: out/minikube delete -p running-upgrade-20210302000143-3349650: (2.68652271s)
--- PASS: TestRunningBinaryUpgrade (97.68s)
PASS
Tests completed in 1m37.681851438s (result code 0)
ok      k8s.io/minikube/test/integration        97.713s
```
```
❯ env TEST_ARGS="-minikube-start-args=--driver=docker -test.run TestStoppedBinaryUpgrade" make integration
go test -ldflags="-X k8s.io/minikube/pkg/version.version=v1.18.0-beta.0 -X k8s.io/minikube/pkg/version.isoVersion=v1.18.0-beta.0 -X k8s.io/minikube/pkg/version.gitCommitID="4903ede887ea788fabe6eb3e9376bb6b5c34f5ba-dirty" -X k8s.io/minikube/pkg/version.storageProvisionerVersion=v4" -v -test.timeout=90m ./test/integration --tags="integration " -minikube-start-args=--driver=docker -test.run TestStoppedBinaryUpgrade 2>&1 | tee "./out/testout_4903ede88.txt"
Found 16 cores, limiting parallelism with --test.parallel=9
=== RUN   TestStoppedBinaryUpgrade
=== PAUSE TestStoppedBinaryUpgrade
=== CONT  TestStoppedBinaryUpgrade
    version_upgrade_test.go:180: (dbg) Run:  /tmp/minikube-v1.8.0.211279187.exe start -p stopped-upgrade-20210302000332-3365124 --memory=2200 --vm-driver=docker
    version_upgrade_test.go:180: (dbg) Done: /tmp/minikube-v1.8.0.211279187.exe start -p stopped-upgrade-20210302000332-3365124 --memory=2200 --vm-driver=docker: (2m25.567023474s)
    version_upgrade_test.go:189: (dbg) Run:  /tmp/minikube-v1.8.0.211279187.exe -p stopped-upgrade-20210302000332-3365124 stop
    version_upgrade_test.go:189: (dbg) Done: /tmp/minikube-v1.8.0.211279187.exe -p stopped-upgrade-20210302000332-3365124 stop: (11.283325788s)
    version_upgrade_test.go:195: (dbg) Run:  out/minikube start -p stopped-upgrade-20210302000332-3365124 --memory=2200 --alsologtostderr -v=1 --driver=docker
    version_upgrade_test.go:195: (dbg) Done: out/minikube start -p stopped-upgrade-20210302000332-3365124 --memory=2200 --alsologtostderr -v=1 --driver=docker: (57.633513774s)
=== RUN   TestStoppedBinaryUpgrade/MinikubeLogs
    version_upgrade_test.go:202: (dbg) Run:  out/minikube logs -p stopped-upgrade-20210302000332-3365124
    version_upgrade_test.go:202: (dbg) Done: out/minikube logs -p stopped-upgrade-20210302000332-3365124: (2.278359114s)
=== CONT  TestStoppedBinaryUpgrade
    helpers_test.go:171: Cleaning up "stopped-upgrade-20210302000332-3365124" profile ...
    helpers_test.go:174: (dbg) Run:  out/minikube delete -p stopped-upgrade-20210302000332-3365124
    helpers_test.go:174: (dbg) Done: out/minikube delete -p stopped-upgrade-20210302000332-3365124: (2.501208049s)
--- PASS: TestStoppedBinaryUpgrade (223.92s)
    --- PASS: TestStoppedBinaryUpgrade/MinikubeLogs (2.28s)
PASS
Tests completed in 3m43.916946831s (result code 0)
ok      k8s.io/minikube/test/integration        223.947s
```
```
❯ env TEST_ARGS="-minikube-start-args=--driver=docker -test.run TestKubernetesUpgrade" make integration
go test -ldflags="-X k8s.io/minikube/pkg/version.version=v1.18.0-beta.0 -X k8s.io/minikube/pkg/version.isoVersion=v1.18.0-beta.0 -X k8s.io/minikube/pkg/version.gitCommitID="4903ede887ea788fabe6eb3e9376bb6b5c34f5ba-dirty" -X k8s.io/minikube/pkg/version.storageProvisionerVersion=v4" -v -test.timeout=90m ./test/integration --tags="integration " -minikube-start-args=--driver=docker -test.run TestKubernetesUpgrade 2>&1 | tee "./out/testout_4903ede88.txt"
Found 16 cores, limiting parallelism with --test.parallel=9
=== RUN   TestKubernetesUpgrade
=== PAUSE TestKubernetesUpgrade
=== CONT  TestKubernetesUpgrade
    version_upgrade_test.go:218: (dbg) Run:  out/minikube start -p kubernetes-upgrade-20210302000729-3386156 --memory=2200 --kubernetes-version=v1.14.0 --alsologtostderr -v=1 --driver=docker
    version_upgrade_test.go:218: (dbg) Done: out/minikube start -p kubernetes-upgrade-20210302000729-3386156 --memory=2200 --kubernetes-version=v1.14.0 --alsologtostderr -v=1 --driver=docker: (1m19.774281973s)
    version_upgrade_test.go:223: (dbg) Run:  out/minikube stop -p kubernetes-upgrade-20210302000729-3386156
    version_upgrade_test.go:223: (dbg) Done: out/minikube stop -p kubernetes-upgrade-20210302000729-3386156: (10.868242281s)
    version_upgrade_test.go:228: (dbg) Run:  out/minikube -p kubernetes-upgrade-20210302000729-3386156 status --format={{.Host}}
    version_upgrade_test.go:228: (dbg) Non-zero exit: out/minikube -p kubernetes-upgrade-20210302000729-3386156 status --format={{.Host}}: exit status 7 (67.396339ms)

        -- stdout --
                Stopped

        -- /stdout --
    version_upgrade_test.go:230: status error: exit status 7 (may be ok)
    version_upgrade_test.go:239: (dbg) Run:  out/minikube start -p kubernetes-upgrade-20210302000729-3386156 --memory=2200 --kubernetes-version=v1.20.5-rc.0 --alsologtostderr -v=1 --driver=docker
    version_upgrade_test.go:239: (dbg) Done: out/minikube start -p kubernetes-upgrade-20210302000729-3386156 --memory=2200 --kubernetes-version=v1.20.5-rc.0 --alsologtostderr -v=1 --driver=docker: (2m57.632182881s)
    version_upgrade_test.go:244: (dbg) Run:  kubectl --context kubernetes-upgrade-20210302000729-3386156 version --output=json
    version_upgrade_test.go:263: Attempting to downgrade Kubernetes (should fail)
    version_upgrade_test.go:265: (dbg) Run:  out/minikube start -p kubernetes-upgrade-20210302000729-3386156 --memory=2200 --kubernetes-version=v1.14.0 --driver=docker
    version_upgrade_test.go:265: (dbg) Non-zero exit: out/minikube start -p kubernetes-upgrade-20210302000729-3386156 --memory=2200 --kubernetes-version=v1.14.0 --driver=docker: exit status 106 (50.569124ms)

        -- stdout --
                * [kubernetes-upgrade-20210302000729-3386156] minikube v1.18.0-beta.0 on Opensuse-Tumbleweed



        -- /stdout --
        ** stderr **
                X Exiting due to K8S_DOWNGRADE_UNSUPPORTED: Unable to safely downgrade existing Kubernetes v1.20.5-rc.0 cluster to v1.14.0
                * Suggestion:

                    1) Recreate the cluster with Kubernetes 1.14.0, by running:

                    minikube delete -p kubernetes-upgrade-20210302000729-3386156
                    minikube start -p kubernetes-upgrade-20210302000729-3386156 --kubernetes-version=v1.14.0

                    2) Create a second cluster with Kubernetes 1.14.0, by running:

                    minikube start -p kubernetes-upgrade-20210302000729-33861562 --kubernetes-version=v1.14.0

                    3) Use the existing cluster at version Kubernetes 1.20.5-rc.0, by running:

                    minikube start -p kubernetes-upgrade-20210302000729-3386156 --kubernetes-version=v1.20.5-rc.0


        ** /stderr **
    version_upgrade_test.go:269: Attempting restart after unsuccessful downgrade
    version_upgrade_test.go:271: (dbg) Run:  out/minikube start -p kubernetes-upgrade-20210302000729-3386156 --memory=2200 --kubernetes-version=v1.20.5-rc.0 --alsologtostderr -v=1 --driver=docker
    version_upgrade_test.go:271: (dbg) Done: out/minikube start -p kubernetes-upgrade-20210302000729-3386156 --memory=2200 --kubernetes-version=v1.20.5-rc.0 --alsologtostderr -v=1 --driver=docker: (2m42.410764045s)
    helpers_test.go:171: Cleaning up "kubernetes-upgrade-20210302000729-3386156" profile ...
    helpers_test.go:174: (dbg) Run:  out/minikube delete -p kubernetes-upgrade-20210302000729-3386156
    helpers_test.go:174: (dbg) Done: out/minikube delete -p kubernetes-upgrade-20210302000729-3386156: (4.957036504s)
--- PASS: TestKubernetesUpgrade (435.81s)
PASS
Tests completed in 7m15.811449278s (result code 0)
ok      k8s.io/minikube/test/integration        435.841s
```
```
❯ env TEST_ARGS="-minikube-start-args=--driver=docker -test.run TestMissingContainerUpgrade" make integration
go test -ldflags="-X k8s.io/minikube/pkg/version.version=v1.18.0-beta.0 -X k8s.io/minikube/pkg/version.isoVersion=v1.18.0-beta.0 -X k8s.io/minikube/pkg/version.gitCommitID="4903ede887ea788fabe6eb3e9376bb6b5c34f5ba-dirty" -X k8s.io/minikube/pkg/version.storageProvisionerVersion=v4" -v -test.timeout=90m ./test/integration --tags="integration " -minikube-start-args=--driver=docker -test.run TestMissingContainerUpgrade 2>&1 | tee "./out/testout_4903ede88.txt"
Found 16 cores, limiting parallelism with --test.parallel=9
=== RUN   TestMissingContainerUpgrade
=== PAUSE TestMissingContainerUpgrade
=== CONT  TestMissingContainerUpgrade
    version_upgrade_test.go:305: (dbg) Run:  /tmp/minikube-v1.9.1.402326120.exe start -p missing-upgrade-20210302001507-3424446 --memory=2200 --driver=docker
    version_upgrade_test.go:305: (dbg) Done: /tmp/minikube-v1.9.1.402326120.exe start -p missing-upgrade-20210302001507-3424446 --memory=2200 --driver=docker: (49.33742475s)
    version_upgrade_test.go:314: (dbg) Run:  docker stop missing-upgrade-20210302001507-3424446
    version_upgrade_test.go:314: (dbg) Done: docker stop missing-upgrade-20210302001507-3424446: (2.700285264s)
    version_upgrade_test.go:319: (dbg) Run:  docker rm missing-upgrade-20210302001507-3424446
    version_upgrade_test.go:325: (dbg) Run:  out/minikube start -p missing-upgrade-20210302001507-3424446 --memory=2200 --alsologtostderr -v=1 --driver=docker
    version_upgrade_test.go:325: (dbg) Done: out/minikube start -p missing-upgrade-20210302001507-3424446 --memory=2200 --alsologtostderr -v=1 --driver=docker: (1m4.557175947s)
    helpers_test.go:171: Cleaning up "missing-upgrade-20210302001507-3424446" profile ...
    helpers_test.go:174: (dbg) Run:  out/minikube delete -p missing-upgrade-20210302001507-3424446
    helpers_test.go:174: (dbg) Done: out/minikube delete -p missing-upgrade-20210302001507-3424446: (4.077126559s)
--- PASS: TestMissingContainerUpgrade (125.62s)
PASS
Tests completed in 2m5.623542774s (result code 0)
ok      k8s.io/minikube/test/integration        125.652s
```

### CODE REVIEW

**Comment by medyagh** [comment]
would u plz rebase the PR I think the go mod diff would go away @prezha

**Review by medyagh — APPROVED** [review]

**Comment by medyagh** [comment]
@prezha one more rebase plz

**Review by ohbus — APPROVED** [review]

**Comment by prezha** [comment]
> @prezha one more rebase plz

@medyagh  sure, no problem, just did

**Comment by medyagh** [comment]
hm...something wrong with the tests ...

File: go.mod [hunk 1]
```diff
@@ -109,28 +102,30 @@ replace (
...
-	k8s.io/kube-scheduler => k8s.io/kube-scheduler v0.17.3
-	k8s.io/kubectl => k8s.io/kubectl v0.17.3
-	k8s.io/kubelet => k8s.io/kubelet v0.17.3
-	k8s.io/legacy-cloud-providers => k8s.io/legacy-cloud-providers v0.17.3
-	k8s.io/metrics => k8s.io/metrics v0.17.3
-	k8s.io/node-api => k8s.io/node-api v0.17.3
-	k8s.io/sample-apiserver => k8s.io/sample-apiserver v0.17.3
-	k8s.io/sample-cli-plugin => k8s.io/sample-cli-plugin v0.17.3
-	k8s.io/sample-controller => k8s.io/sample-controller v0.17.3
+	k8s.io/api => k8s.io/api v0.20.5
```
Comment (medyagh) [inline go.mod:105]: do we still need these Replaces ? what happens if we get rid of the replaces?

**Comment by medyagh** [comment]
> **kvm2 Driver**
> Times for minikube: 68.1s 66.8s 67.8s
> Average time for minikube: 67.6s
> 
> Times for Minikube (PR 10683): 68.2s 65.6s 66.2s
> Average time for Minikube (PR 10683): 66.7s
> 
> Averages Time Per Log
> 
> **docker Driver**
> Times for minikube: 31.7s 31.8s 29.3s
> Average time for minikube: 30.9s
> 
> Times for Minikube (PR 10683): 42.3s 32.8s 38.8s
> Average time for Minikube (PR 10683): 38.0s
> 
> Averages Time Per Log

The metrics dont look very good for docker driver ... and the number of failures is higher than normal on docker linux

**Pull request merged**